mne/preprocessing/faster.py

```diff
@@ -0,0 +1,368 @@
+        List of metrics to use. Can be any combination of:
+            'amplitude', 'variance', 'deviation'
+        Defaults to all of them.
+
+    Returns
+    -------
+    bads : list of int
+        The indices of the bad epochs.
+    """
+
+    metrics = {
+        'amplitude': lambda x: np.mean(np.ptp(x, axis=2), axis=1),
+        'deviation': lambda x: np.mean(_deviation(x), axis=1),
+        'variance':  lambda x: np.mean(np.var(x, axis=2), axis=1),
+    }
+
+    if picks is None:
+        picks = pick_types(epochs.info, meg=False, eeg=True, exclude='bads')
+    if use_metrics is None:
+        use_metrics = metrics.keys()
+
+    data = epochs.get_data()[:, picks, :]
+
+    bads = []
+    for m in use_metrics:
+        s = metrics[m](data)
+        b = find_outliers(s, thres)
+        logger.info('Bad by %s:\n\t%s' % (m, b))
+        bads.append(b)
+
+    return np.unique(np.concatenate(bads)).tolist()
+
+
+def faster_bad_components(ica, epochs, thres=3, use_metrics=None):
+    """Implements the third step of the FASTER algorithm.
+    
+    This function attempts to automatically mark bad ICA components by
+    performing outlier detection.
+
+    Parameters
+    ----------
+    ica : Instance of ICA
+        The ICA operator, already fitted to the supplied Epochs object.
+    epochs : Instance of Epochs
+        The untransformed epochs to analyze.
+    thres : float
+        The threshold value, in standard deviations, to apply. A component
+        crossing this threshold value is marked as bad. Defaults to 3.
+    use_metrics : list of str
+        List of metrics to use. Can be any combination of:
+            'eog_correlation', 'kurtosis', 'power_gradient', 'hurst',
+            'median_gradient'
+        Defaults to all of them.
+
+    Returns
+    -------
+    bads : list of int
+        The indices of the bad components.
+
+    See also
+    --------
+    ICA.find_bads_ecg
+    ICA.find_bads_eog
+    """
+    source_data = ica.get_sources(epochs).get_data().transpose(1,0,2)
+    source_data = source_data.reshape(source_data.shape[0], -1)
+
+    metrics = {
+        'eog_correlation': lambda x: x.find_bads_eog(epochs)[1],
+        'kurtosis':        lambda x: kurtosis(
+                               np.dot(
+                                   x.mixing_matrix_.T,
+                                   x.pca_components_[:x.n_components_]),
+                               axis=1),
+        'power_gradient':  lambda x: _power_gradient(source_data,
+                                                     x.info['sfreq']),
+        'hurst':           lambda x: _hurst(source_data),
+        'median_gradient': lambda x: np.median(np.abs(np.diff(source_data)),
+                                               axis=1),
+        'line_noise':      lambda x: _freqs_power(source_data,
+                                                  epochs.info['sfreq'],
+                                                  [50, 60]),
+    }
```

Comment:
btw there are a couple of pep8 errors, e.g. see white space after keys in the dicts